core/src/main/java/io/questdb/griffin/engine/functions/eq/EqUuidStrFunctionFactory.java

```diff
@@ -0,0 +1,52 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2022 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin.engine.functions.eq;
+
+import io.questdb.cairo.CairoConfiguration;
+import io.questdb.cairo.sql.Function;
+import io.questdb.griffin.FunctionFactory;
+import io.questdb.griffin.SqlException;
+import io.questdb.griffin.SqlExecutionContext;
+import io.questdb.std.IntList;
+import io.questdb.std.ObjList;
+
+public final class EqUuidStrFunctionFactory implements FunctionFactory {
+    @Override
+    public String getSignature() {
+        return "=(ZS)";
```

Comment:
i'm not sure this is necessary. If casting is correct, string can be cast to UUID and (ZZ) will execute? Can we verify this ?

Reply:
Casting will work the other way around: It'll cast the UUID to String. Which is inefficient for comparison. We can assume a UUID column will be often compared against a String constant. (=filter). 

Why is it casting UUID to String and not the other way around? Because you can always cast any UUID to String: All UUIDs can always be represented as a String. but not the other way around. 

In other words: It works without this function - by casting UUID to String - but it's inefficient. This is basically an optimization.